packages/homeserver/src/controllers/federation/profiles.controller.ts

```diff
@@ -69,32 +74,25 @@ export const profilesPlugin = (app: Elysia) => {
 		)
 		.get(
 			'/_matrix/federation/v1/make_join/:roomId/:userId',
-			async ({ params, query }) => {
-				const response = await profilesService.makeJoin(
-					params.roomId,
-					params.userId,
-					query.ver,
-				);
-				return {
-					room_version: response.room_version,
-					event: {
-						...response.event,
-						content: {
-							...response.event.content,
-							membership: 'join',
+			async ({ params, query: _query }) => {
+				const { roomId, userId } = params;
+
+				const roomInformation = await stateService.getRoomInformation(roomId);
 
-							join_authorised_via_users_server:
-								response.event.content.join_authorised_via_users_server,
-						},
-						room_id: response.event.room_id,
-						sender: response.event.sender,
+				const membershipEvent = PersistentEventFactory.newMembershipEvent(
+					roomId,
+					userId,
+					userId,
+					'join',
+					roomInformation,
+				);
 
-						state_key: response.event.state_key,
+				await stateService.addAuthEvents(membershipEvent);
+				await stateService.addPrevEvents(membershipEvent);
```

Comment:
This needs to be inside the service, otherwise we need to replicate the same logic on RC.